src/app-layer-enip.c

```diff
@@ -0,0 +1,656 @@
+/* Copyright (C) 2015 Open Information Security Foundation
+ *
+ * You can copy, redistribute or modify this Program under the terms of
+ * the GNU General Public License version 2 as published by the Free
+ * Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * version 2 along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
+ */
+
+/**
+ * \file
+ *
+ * \author Kevin Wong <kwong@solananetworks.com>
+ *
+ * App-layer parser for ENIP protocol
+ *
+ */
+
+#include "suricata-common.h"
+
+#include "util-debug.h"
+#include "util-byte.h"
+#include "util-enum.h"
+#include "util-mem.h"
+#include "util-misc.h"
+
+#include "stream.h"
+
+#include "app-layer-protos.h"
+#include "app-layer-parser.h"
+#include "app-layer-enip.h"
+#include "app-layer-enip-common.h"
+
+#include "app-layer-detect-proto.h"
+
+#include "conf.h"
+#include "decode.h"
+
+#include "detect-parse.h"
+#include "detect-engine.h"
+#include "util-byte.h"
+#include "util-unittest.h"
+#include "util-unittest-helper.h"
+#include "pkt-var.h"
+#include "util-profiling.h"
+
+
+SCEnumCharMap enip_decoder_event_table[ ] = {
+    { NULL,                         -1 },
+};
+
+/** \brief get value for 'complete' status in ENIP
+ *
+ *  For ENIP we use a simple bool.
+ */
+int ENIPGetAlstateProgress(void *tx, uint8_t direction)
+{
+
+   // printf("ENIPGetAlstateProgress direction %d\n", direction);
+
+    return 1;
+}
+
+/** \brief get value for 'complete' status in ENIP
+ *
+ *  For ENIP we use a simple bool.
+ */
+int ENIPGetAlstateProgressCompletionStatus(uint8_t direction)
+{
+  //  printf("ENIPGetAlstateProgressCompletionStatus direction %d\n", direction);
+
+    return 1;
+}
+
+
+
+DetectEngineState *ENIPGetTxDetectState(void *vtx)
+{
+    ENIPTransaction *tx = (ENIPTransaction *)vtx;
+    return tx->de_state;
+}
+
+int ENIPSetTxDetectState(void *state, void *vtx, DetectEngineState *s)
+{
+    ENIPTransaction *tx = (ENIPTransaction *)vtx;
+    tx->de_state = s;
+    return 0;
+}
+
+
+
+void *ENIPGetTx(void *alstate, uint64_t tx_id) {
+    ENIPState         *enip = (ENIPState *) alstate;
+    ENIPTransaction   *tx = NULL;
+
+    if (enip->curr && enip->curr->tx_num == tx_id + 1)
+        return enip->curr;
+
+    TAILQ_FOREACH(tx, &enip->tx_list, next) {
+        SCLogDebug("tx->tx_num %"PRIu64", tx_id %"PRIu64, tx->tx_num, (tx_id+1));
```

Comment:
<pre>
  CC       app-layer-enip.o
app-layer-enip.c:108:59: error: format specifies type 'unsigned long' but the argument has type 'uint16_t' (aka 'unsigned short') [-Werror,-Wformat]
        SCLogDebug("tx->tx_num %"PRIu64", tx_id %"PRIu64, tx->tx_num, (tx_id+1));
                                                          ^~~~~~~~~~
./util-debug.h:335:51: note: expanded from macro 'SCLogDebug'
#define SCLogDebug(...)       SCLog(SC_LOG_DEBUG, __VA_ARGS__)
                                                  ^
./util-debug.h:212:59: note: expanded from macro 'SCLog'
            snprintf(_sc_log_msg, SC_LOG_MAX_LOG_MSG_LEN, __VA_ARGS__);         \
                                                          ^
1 error generated.
</pre>